smartsim/entity/model.py

```diff
@@ -343,18 +344,12 @@ def colocate_db_tcp(
 
     def _set_colocated_db_settings(
         self,
-        connection_options: t.Mapping[str, t.Union[int, t.List[str], str]],
-        common_options: t.Dict[
+        connection_options: Mapping[str, int | list[str] | str],
+        common_options: dict[
             str,
-            t.Union[
-                t.Union[t.Iterable[t.Union[int, t.Iterable[int]]], None],
-                bool,
-                int,
-                str,
-                None,
-            ],
+            Iterable[int | Iterable[int]] | None | bool | int | str | None,
```

Comment:
Nit: This is a problem with the original type hint also, but in theory there is no need to specify `| None` more than once:
```suggestion
            Iterable[int | Iterable[int]] | bool | int | str | None,
```